### PR TITLE
Filter validation issue

### PR DESCRIPTION
**Pull request by tpluscode**
https://user-images.githubusercontent.com/588128/204266371-13c891df-d61f-4822-b8d7-cd027c53958f.mp4

### CODE REVIEW

File: apps/www/src/state/plugin/view.js [hunk 1]
```diff
@@ -27,16 +31,28 @@ export const viewForm = {
...
-      setViewValidity(state, validityReport) {
-        return { ...state, validityReport }
+      setViewValidity(state, { report, shapes, pointer }) {
+        return { ...state, validity: { report, shapes, pointer } }
       },
     },
     effects(store) {
       const dispatch = store.getDispatch()
 
+      function whenValid(effect) {
```
Comment (cristianvasquez) [inline apps/www/src/state/plugin/view.js:44]: perhaps 'checkValidity' or 'whenValidity' ? as it could be not valid...

Comment (tpluscode) [reply apps/www/src/state/plugin/view.js:44]: The intended semantics is "when valid, call `effect`".

File: packages/view-util/lib/dimensions.js [hunk 2]
```diff
@@ -51,3 +51,19 @@ export async function populateDimensionIdentifiers(pointer, metaLookup) {
       }
     })
 }
+
+export function setViewAs(view) {
```
Comment (cristianvasquez) [inline packages/view-util/lib/dimensions.js:55]: I would add some cues in the function name,  'setViewAsProp' as it can be read as 'set view as view'

**Review by cristianvasquez — APPROVED** [review]
No major comments on this one!

File: apps/www/src/state/plugin/view.js (hunk 1 above)
Comment (cristianvasquez) [reply apps/www/src/state/plugin/view.js:44]: then what about 'whenValidCall' ?

**Pull request merged**